### ISSUE TITLE
Improve import of custom yaml

### ISSUE THREAD

**Issue by zackbatist** [dendronhq/dendron#659]
## Please Select if your Request is Either something new or an Enhancement ##

- [x] Enhancement of an existing Feature.
- [ ] Request of a new feature.

## Please select the area your request applys to. (Multiple selections are Possible) ##

- [x] Onboard. Initial vault setup and import/export pods 
- [ ] Create. Note creation, lookup, snippets and templates
- [ ] Retrieve. Backlinks, references, graph view
- [ ] Structure. Refactoring, multi-vault and schemas
- [ ] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Is your feature request related to a problem? Please describe ##
I have a bunch of custom front matter in notes that I would like to import which break dendron doctor.

## Describe the solution you'd like ##

I would like to be able to select which existing front matter variables to import, and map them onto new front matter variables in my dendron system. Notes that have the existing front matter specified will have the values from that variable copied as values of the new variables. For example:

| old files      | new files |
| ----------- | ----------- |
| `pet: puppy`      | `animal: puppy`       |
| `pet: kitty`   | `animal: kitty`        |

Complementary and perhaps also necessary for this to work, schemas may specify front matter to be automatically generated (or at least the variables with a blank value) for new notes under the specified branch.

A tangential extension of this feature request might be that a pod could do a selective import, i.e. import only notes containing a specific yaml value. For example, I could specify I want to import only notes containing `pet: 'puppy'`, or anything with the `pet` variable by using `pet: *`.

Some of this may lead to other issues relating to the ability to refactor yaml headers, sorry if this is jumping the gun.

## Describe alternatives you've considered ##

When importing notes, I'm considering including the custom yaml metadata under a secondary front matter frame. Kind of like this:

```
---
---       <-- this will be removed after running dendron doctor
pet: 'puppy'
colour: 'red'
---
```

This would import the top level metadata as blank, and therefore not break dendron doctor. Then I would run dendron doctor, and do a mass find and replace to modify `pet:` into `animal`. After all this, I would smush together the secondary yaml frame into the primary frame by removing the second set of `---`.

I haven't tried any of this yet. I'm hoping others might chime in with potentially simpler solutions.

## Additional context ##
N/A

**Comment by kpathakota** [comment]
Yeah, there are a couple work items in this one issue that have the potential of adding a ton of value to our existing frontmatter feature. : 
1. Have the markdown import pod preserve frontmatter. 
2. Frontmatter "mapping" during markdown import -> transform existing elements to other elements. 
3. Schemas specify custom-frontmatter per-hierarchy (get auto-populated when creating notes under that hierarchy). 

Overall, I think we can build this feature iteratively starting with the first one. CC: @kevinslin

**Comment by kevinslin** [comment]
I agree. We can take on 1. for next week and have the other two float in the backlog
